src/pages/Reviews/SingleReview.jsx

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useState } from "react";
+import { useLoaderData, useNavigate } from "react-router-dom";
+import reqURLs from "../../api";
+import { NotFound } from "../NotFound/NotFound";
+import { Helmet, HelmetProvider } from "react-helmet-async";
+
+export default function SingleReview() {
+  const navigate = useNavigate();
+  const [user, setUser] = useState("");
+
+  const { review } = useLoaderData();
+
+  useEffect(() => {
+    const getUser = async () => {
+      const response = await fetch(`${reqURLs.APIUsers}`);
+      const { users } = await response.json();
+      const author = users.filter((user) => user.username === review.owner);
+      setUser(author[0]);
+    };
+
+    if (!review) {
+      navigate("/not-found", { state: { from: "single-review" } });

```

Comment:
I really like the use of navigate here to move to the not found page if there's no review back from the api